### DISCUSSION TITLE
Fallback for disabled webgl

### DISCUSSION THREAD

**Discussion by benjaminpreiss** [maplibre/maplibre-gl-js#4473] [Q&A]
With mapbox gl js there is a fallback in case webgl is not available: https://docs.mapbox.com/help/tutorials/use-mapbox-js-as-fallback/

Is this also possible with maplibre gl js? If so, how? Can you point me in the right direction?

**Comment by HarelM** [comment]
You might be able to use mapbox.js, but you'll need a server to serve rasters of your vector tiles I believe.
This does not come out of the box in maplibre in any form, sorry...

**Reply by benjaminpreiss** [reply]
We use protomaps currently anyway (https://protomaps.com/) to serve tiles - does that count?

**Reply by benjaminpreiss** [reply]
And I believe we still need to use a mapbox accessToken and are therefore bound to mapbox? https://docs.mapbox.com/help/tutorials/use-mapbox-js-as-fallback/#initialize-a-map-with-mapboxjs

**Comment by IvanSanchez** [comment]
You might be able to detect WebGL support and then fallback to a map with e.g. Leaflet (which is WebGL-less). The challenge lies in displaying your data with a similar graphical appearance, since (to my knowledge) vector stylesheet support for renderers based solely in 2D Canvas is... lacking.

**Reply by benjaminpreiss** [reply]
I have made the same experience with 2D canvas based renderers - which is a pity!

**Reply by benjaminpreiss** [reply]
This looks promising tho? https://github.com/maplibre/maplibre-gl-leaflet?tab=readme-ov-file
